NitroxServer/Communication/Packets/Processors/FootstepPacketProcessor.cs

```diff
@@ -0,0 +1,52 @@
+using NitroxModel.DataStructures.Unity;
+using NitroxModel.Packets;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using NitroxServer.GameLogic;
+
+namespace NitroxServer.Communication.Packets.Processors
+{
+    public class FootstepPacketProcessor : AuthenticatedPacketProcessor<FootstepPacket>
+    {
+        private readonly float footstepAudioRange = 20f;
+        private readonly PlayerManager playerManager;
+
+        public FootstepPacketProcessor(PlayerManager playerManager)
+        {
+            this.playerManager = playerManager;
+        }
+
+        public override void Process(FootstepPacket footstepPacket, Player sendingPlayer)
+        {
+            var players = playerManager.GetAllPlayers();
+            foreach (Player player in players)
+            {
+                if (sendingPlayer.SubRootId.HasValue)
```

Comment:
The first if in this brnaching tree should be the most common. In this case the `NitroxVector3.Distance [...] && player != sendingPlayer` bc it's in both branches

Reply:
Done